gooddata-sdk/gooddata_sdk/catalog/workspace/entity_model/graph_objects/graph.py

```diff
@@ -0,0 +1,95 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import List, Optional, Type
+
+import attr
+import cattr
+from cattrs.gen import make_dict_structure_fn, make_dict_unstructure_fn, override
+
+from gooddata_metadata_client.model.dependent_entities_edge import DependentEntitiesEdge
+from gooddata_metadata_client.model.dependent_entities_graph import DependentEntitiesGraph
+from gooddata_metadata_client.model.dependent_entities_node import DependentEntitiesNode
+from gooddata_metadata_client.model.dependent_entities_request import DependentEntitiesRequest
+from gooddata_metadata_client.model.dependent_entities_response import DependentEntitiesResponse
+from gooddata_metadata_client.model.entity_identifier import EntityIdentifier
+from gooddata_sdk.catalog.base import Base
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class CatalogDependentEntitiesRequest(Base):
+    identifiers: List[CatalogEntityIdentifier] = attr.field(factory=list)
+
+    @staticmethod
+    def client_class() -> Type[DependentEntitiesRequest]:
+        return DependentEntitiesRequest
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class CatalogDependentEntitiesResponse(Base):
+    graph: CatalogDependentEntitiesGraph
+
+    @staticmethod
+    def client_class() -> Type[DependentEntitiesResponse]:
+        return DependentEntitiesResponse
+
+    @staticmethod
+    def _custom_converter() -> Optional[cattr.GenConverter]:
```

Comment:
This looks like converter is registered only for `CatalogDependentEntitiesResponse` but I could ask for conversion of nested structure, like `CatalogDependentEntitiesGraph` or `CatalogDependentEntitiesEdge`.

Ideally the implementation should:
1. go through attribute metadata and build `hooks`
2. register hooks from all its attributes to converter instance

Step (1)  must be generic, step (2) ideally too (requires typing introspection). For simplicity, step(2) can be done manually in case nested structures are in use.

Reply:
That is a fair point. The issue could be resolved using the singleton you proposed in the previous comment. I will make a draft.